src/pybind/mgr/cephadm/serve.py

```diff
@@ -1091,10 +1091,11 @@ def _check_daemons(self) -> None:
                 self.log.debug(f'{dd.name()} deps {last_deps} -> {deps}')
                 self.log.info(f'Reconfiguring {dd.name()} (dependencies changed)...')
                 action = 'reconfig'
-                # we need only redeploy if secure_monitoring_stack value has changed:
+                # we need only redeploy if secure_monitoring_stack or admin-gateway value has changed:
+                # TODO(redo): check if we should just go always with redeploy (it's fast enough)
                 if dd.daemon_type in ['prometheus', 'node-exporter', 'alertmanager']:
-                    diff = list(set(last_deps) - set(deps))
-                    if any('secure_monitoring_stack' in e for e in diff):
+                    diff = list(set(last_deps).symmetric_difference(set(deps)))
+                    if any(('secure_monitoring_stack' in e) or ('admin-gateway' in e) for e in diff):
```

Comment:
well, not for this basic functionality but we will need to do when mTLS is merged because the config file needs some adjustment to include certificates needed for mTLS.